kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PageToken.java

```diff
@@ -0,0 +1,164 @@
+package io.delta.kernel.internal.replay;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.data.GenericRow;
+import io.delta.kernel.types.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Page Token Class for Pagination Support */
+public class PageToken {
+
+  public static PageToken fromRow(Row row) {
+    requireNonNull(row);
+    if (!PAGE_TOKEN_SCHEMA.equals(row.getSchema())) {
+      throw new IllegalArgumentException(
+          "Invalid Page Token: input row schema does not match expected PageToken schema. "
+              + "Expected: "
+              + PAGE_TOKEN_SCHEMA
+              + ", Got: "
+              + row.getSchema());
+    }
+
+    for (int i = 0; i < PAGE_TOKEN_SCHEMA.length(); i++) {
+      if (row.isNullAt(i) && !Objects.equals(PAGE_TOKEN_SCHEMA.at(i).getName(), "sidecarIndex")) {
+        throw new IllegalArgumentException(
+            "Invalid Page Token: required field '"
+                + PAGE_TOKEN_SCHEMA.at(i).getName()
+                + "' is null at index "
+                + i);
+      }
+    }
+
+    return new PageToken(
+        row.getString(0), // logFileName
+        row.getLong(1), // rowIndexInFile
+        Optional.ofNullable(row.isNullAt(2) ? null : row.getLong(2)), // sidecarIndex
+        row.getString(3), // kernelVersion
+        row.getString(4), // tablePath
+        row.getLong(5), // tableVersion
+        row.getLong(6), // predicateHash
+        row.getLong(7)); // logSegmentHash
+  }
+
+  /** Schema for PageToken Row representation */
+  public static final StructType PAGE_TOKEN_SCHEMA =
+      new StructType()
+          .add("logFileName", StringType.STRING)
+          .add("rowIndexInFile", LongType.LONG)
+          .add("sidecarIndex", LongType.LONG)
+          .add("kernelVersion", StringType.STRING)
+          .add("tablePath", StringType.STRING)
+          .add("tableVersion", LongType.LONG)
+          .add("predicateHash", LongType.LONG)
+          .add("logSegmentHash", LongType.LONG);
+
+  // ===== Variables to mark where the last page ended (and the current page starts) =====
+  /**
+   * The name of the log file where the current page starts. This is the same as the last log file
```

Comment:
will this always be the case?
what if we are able to detect that this columnar batch is the **last** columnar batch in the file?
will we then set this `startingLogFileName` to the **next** file?